### DISCUSSION TITLE
Escaping characters in theme.yaml

### DISCUSSION THREAD

**Discussion by VoxLuna** [mathoudebine/turing-smart-screen-python#403] [Q&A] [answered]
I'm trying to define a text string inside theme.yaml for a Windows drive letter, as in C:

```
DISK_USAGE:
    TEXT: C':' 2TB NVMe          
    X: 42
    Y: 419
```

The colon causes theme-editor and configure.py to crash when it's used as above.  Normally you'd escape it; the usual way is to surround it with double-quotes.  But doing so will display as:

![turing-colon](https://github.com/mathoudebine/turing-smart-screen-python/assets/116472/121c47a8-ba99-49f3-a905-06d406cc7aff)

Maybe because it's not Python code but YAML.  I thought about searching for a glyph representing "colon" but that doesn't solve the problem of, how can I compensate for such characters so they are displayed but not interpreted?

**Comment by Bru-Katsu** [comment, marked as answer]
> I'm trying to define a text string inside theme.yaml for a Windows drive letter, as in C:
> 
> ```
> DISK_USAGE:
>     TEXT: C':' 2TB NVMe          
>     X: 42
>     Y: 419
> ```
> 
> The colon causes theme-editor and configure.py to crash when it's used as above. Normally you'd escape it; the usual way is to surround it with double-quotes. But doing so will display as:
> 
> ![turing-colon](https://private-user-images.githubusercontent.com/116472/286743757-121c47a8-ba99-49f3-a905-06d406cc7aff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE4MjQ2MzQsIm5iZiI6MTcwMTgyNDMzNCwicGF0aCI6Ii8xMTY0NzIvMjg2NzQzNzU3LTEyMWM0N2E4LWJhOTktNDlmMy1hOTA1LTA2ZDQwNmNjN2FmZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwNlQwMDU4NTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00M2JiZWQ2MjM0M2MxZTc3YjkyZWI3ZmM0ODMyZjlhNzczOWE2NTJmYmVkMmQ2NGJkOGNkNWJmODk3MWMyOTZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.gF90fhoJaCuamUsS415ant60bTxtj476plc2eiHDImA)
> 
> Maybe because it's not Python code but YAML. I thought about searching for a glyph representing "colon" but that doesn't solve the problem of, how can I compensate for such characters so they are displayed but not interpreted?


Try this:
```
DISK_USAGE:
    TEXT: 'C: 2TB NVMe'
    X: 42
    Y: 419
``` 
![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/30190047/058bef8c-33eb-4dea-9f86-456c6de0e770)
